### PR TITLE
Build for more architectures

### PR DESCRIPTION
**Pull request by jdblischak**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

I purposefully didn't bump the build number. The rerendering didn't affect the `ci_support/` files for the existing builds, so it's not necessary to upload redundant binaries for these platforms

### CODE REVIEW

**Review by Shelnutt2 — APPROVED** [review]

**Comment by jjerphan** [comment]
Thanks ! I think we should bump the build number in case dependencies have been released in the meantime to take them into account.

What do you think?

**Comment by jdblischak** [comment]
> Thanks ! I think we should bump the build number in case dependencies have been released in the meantime to take them into account.
>
> What do you think?

I'm happy to be proven wrong if my mental model is incorrect, but I don't see the value in uploading new binaries for linux-64, osx-64, and win-64 when they were just uploaded 2 days ago in 909285d

It's my understanding that the motivation for bumping the build number whenever one rerenders a recipe is because it's possible one of the pinnings may have changed, which would affect it's hash (and importantly its ability to be co-installed with other conda-forge binaries that follow conda-forge-pinning).

But in this case, the pinnings in the `.ci_support/` YAML files are unchanged, and thus the hash is also unchanged. As an explicit example, here are the pinnings/hash for the linux-64 build in this PR:

```
INFO :: The inputs making up the hashes for the built packages are as follows:
{
  "azure-core-cpp-1.10.2-h91d86a7": {
    "recipe": {
      "channel_targets": "conda-forge main",
      "cxx_compiler": "gxx",
      "cxx_compiler_version": "12",
      "libcurl": "8",
      "openssl": "3",
      "target_platform": "linux-64"
    }
  }
}
```

And here are the pinnings/hash from the [previous build 2 days ago](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=761650&view=logs&j=656edd35-690f-5c53-9ba3-09c10d0bea97&t=986b1512-c876-5f92-0d81-ba851554a0a3) (they are identical)

```
INFO :: The inputs making up the hashes for the built packages are as follows:
{
  "azure-core-cpp-1.10.2-h91d86a7": {
    "recipe": {
      "channel_targets": "conda-forge main",
      "cxx_compiler": "gxx",
      "cxx_compiler_version": "12",
      "libcurl": "8",
      "openssl": "3",
      "target_platform": "linux-64"
    }
  }
}
```

In summary, the goal of this PR is to build binaries for 3 additional target platforms

**Comment by jjerphan** [comment]
> new binaries for linux-64, osx-64, and win-64 when they were just uploaded 2 days ago in 909285d

You are right. In that case, it is safe to merge.

**Review by jjerphan — APPROVED** [review]
